feature-libs/product-configurator/rulebased/components/overview-form/configurator-overview-form.component.spec.ts

```diff
@@ -247,6 +247,105 @@ describe('ConfigurationOverviewFormComponent', () => {
+        },
+        {
+          attribute: 'C1',
+          value: 'V4',
+        },
+      ];
+
+      let result = component.getStyleClasses(attributes, 0);
+      expect(result.includes('general')).toBe(true);
+      expect(result.includes('bundle')).toBe(false);
+      expect(result.includes('margin')).toBe(true);
+      expect(result.includes('last-value-pair')).toBe(false);
+
+      result = component.getStyleClasses(attributes, 1);
+      expect(result.includes('general')).toBe(true);
+      expect(result.includes('bundle')).toBe(false);
+      expect(result.includes('margin')).toBe(false);
+      expect(result.includes('last-value-pair')).toBe(false);
+
+      result = component.getStyleClasses(attributes, 2);
+      expect(result.includes('general')).toBe(true);
+      expect(result.includes('bundle')).toBe(false);
+      expect(result.includes('margin')).toBe(false);
+      expect(result.includes('last-value-pair')).toBe(false);
+
+      result = component.getStyleClasses(attributes, 3);
+      expect(result.includes('general')).toBe(false);
+      expect(result.includes('bundle')).toBe(false);
+      expect(result.includes('margin')).toBe(false);
+      expect(result.includes('last-value-pair')).toBe(true);
+    });
+
+    it('should return certain style classes for bundle attribute type', () => {
+      initialize();
+      const attributes: Configurator.AttributeOverview[] = [
+        {
+          attribute: 'C1',
+          value: 'V1',
+          type: Configurator.AttributeOverviewType.BUNDLE,
+        },
+        {
+          attribute: 'C1',
+          value: 'V2',
+          type: Configurator.AttributeOverviewType.BUNDLE,
+        },
+        {
+          attribute: 'C1',
+          value: 'V3',
+          type: Configurator.AttributeOverviewType.BUNDLE,
+        },
+        {
+          attribute: 'C1',
+          value: 'V4',
+        },
+      ];
+
+      let result = component.getStyleClasses(attributes, 0);
+      expect(result.includes('bundle')).toBe(true);
+      expect(result.includes('general')).toBe(false);
+      expect(result.includes('margin')).toBe(true);
+      expect(result.includes('last-value-pair')).toBe(false);
+
+      result = component.getStyleClasses(attributes, 1);
+      expect(result.includes('bundle')).toBe(true);
+      expect(result.includes('general')).toBe(false);
+      expect(result.includes('margin')).toBe(false);
+      expect(result.includes('last-value-pair')).toBe(false);
+
+      result = component.getStyleClasses(attributes, 2);
+      expect(result.includes('bundle')).toBe(true);
+      expect(result.includes('general')).toBe(false);
+      expect(result.includes('margin')).toBe(false);
+
+      result = component.getStyleClasses(attributes, 3);
+      expect(result.includes('bundle')).toBe(false);
+      expect(result.includes('general')).toBe(false);
+      expect(result.includes('margin')).toBe(false);
+      expect(result.includes('last-value-pair')).toBe(true);
```

Comment:
This are at least 4 tests. Can you please split this tests into several smaller ones, so that the method/feature under test is only called once.

Reply:
Done!